### PR TITLE
feat(web.hosting): backup dbs - add filters and sort by snapshot date

### PR DESCRIPTION
**Pull request by dectotam**
| Question         | Answer
| ---------------- | ---
| Branch?          | master
| Bug fix?         | no
| New feature?     | no
| Breaking change? | no
| Tickets          | Feat MANAGER-12684
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [ ] ~~Only FR translations have been updated~~
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

## Description

Hosting - Backup Databases - Apply a filter and snapshot display order

### CODE REVIEW

File: packages/manager/apps/web/client/app/hosting/database/dump/hosting-database-dump.controller.js [hunk 1]
```diff
@@ -62,24 +74,24 @@ export default class DatabaseDumpsCtrl {
...
   goTo(page, target) {
     this.$window.open(page, target);
   }
 
+  static getSnapshotDateOfDump(dump) {
+    if (!dump.creationDate) {
+      return undefined;
+    }
+
+    const snapshotDate = moment(dump.creationDate);
```
Comment (qpavy) [inline packages/manager/apps/web/client/app/hosting/database/dump/hosting-database-dump.controller.js:86]:
```suggestion
    const snapshotDate = new Date(dump.creationDate);
```

File: packages/manager/apps/web/client/app/hosting/database/dump/hosting-database-dump.controller.js [hunk 2]
```diff
@@ -62,24 +74,24 @@ export default class DatabaseDumpsCtrl {
...
+    if (dump.type === 'daily.1') {
+      snapshotDate.subtract(1, 'd');
```
Comment (qpavy) [inline packages/manager/apps/web/client/app/hosting/database/dump/hosting-database-dump.controller.js:88]:
```suggestion
      snapshotDate.setDate(snapshotDate.getDate() -1);
```

File: packages/manager/apps/web/client/app/hosting/database/dump/hosting-database-dump.controller.js [hunk 3]
```diff
@@ -62,24 +74,24 @@ export default class DatabaseDumpsCtrl {
...
+    } else if (dump.type === 'weekly.1') {
+      snapshotDate.subtract(1, 'w');
```
Comment (qpavy) [inline packages/manager/apps/web/client/app/hosting/database/dump/hosting-database-dump.controller.js:90]:
```suggestion
      snapshotDate.setDate(snapshotDate.getDate() - 7);
```

File: packages/manager/apps/web/client/app/hosting/database/dump/hosting-database-dump.controller.js [hunk 4]
```diff
@@ -62,24 +74,24 @@ export default class DatabaseDumpsCtrl {
...
+    }
+    return snapshotDate.format();
```
Comment (qpavy) [inline packages/manager/apps/web/client/app/hosting/database/dump/hosting-database-dump.controller.js:92]: use Intl to format date

**Review by qpavy — DISMISSED** [review]

File: packages/manager/apps/web/client/app/hosting/database/dump/hosting-database-dump.controller.js [hunk 5]
```diff
@@ -62,22 +76,38 @@ export default class DatabaseDumpsCtrl {
...
 
-  goTo(page, target) {
-    this.$window.open(page, target);
+  static getSnapshotDateOfDump(dump) {
+    if (!dump.creationDate) {
+      return undefined;
+    }
+
+    const snapshotDate = new Date(dump.creationDate);
+    if (dump.type === 'daily.1') {
```
Comment (ghyenne) [inline packages/manager/apps/web/client/app/hosting/database/dump/hosting-database-dump.controller.js:104]: Maybe move the different bump type in constant file.

File: packages/manager/apps/web/client/app/hosting/database/dump/hosting-database-dump.controller.js [hunk 6]
```diff
@@ -62,22 +76,38 @@ export default class DatabaseDumpsCtrl {
...
   }
 
-  goTo(page, target) {
-    this.$window.open(page, target);
+  static getSnapshotDateOfDump(dump) {
+    if (!dump.creationDate) {
+      return undefined;
+    }
+
+    const snapshotDate = new Date(dump.creationDate);
```
Comment (ghyenne) [inline packages/manager/apps/web/client/app/hosting/database/dump/hosting-database-dump.controller.js:103]:
```suggestion
  static getSnapshotDateOfDump(dump) {
  const { creationDate, type } = dump;
    if (!creationDate) {
      return undefined;
    }

    const snapshotDate = new Date(creationDate);
```

**Review by ghyenne — DISMISSED** [review]

File: packages/manager/apps/web/client/app/hosting/database/dump/hosting-database-dump.controller.js [hunk 7]
```diff
@@ -46,12 +52,23 @@ export default class DatabaseDumpsCtrl {
...
+      .$promise.then(({ data }) => {
+        this.databaseDumps = data
+          .map((item) => {
+            return {
+              ...item,
+              ...{
+                snapshotDate: this.constructor.getSnapshotDateOfDump(item),
+              },
+            };
+          })
```
Comment (JacquesLarique) [inline packages/manager/apps/web/client/app/hosting/database/dump/hosting-database-dump.controller.js:70]:
```suggestion
          .map((item) => ({
              ...item,
              snapshotDate: this.constructor.getSnapshotDateOfDump(item),
          }))
```

File: packages/manager/apps/web/client/app/hosting/database/dump/hosting-database-dump.controller.js [hunk 8]
```diff
@@ -46,12 +52,23 @@ export default class DatabaseDumpsCtrl {
   loadDumps() {
     this.databaseDumps = undefined;
 
-    return this.hostingDatabase
-      .getDumpIds(this.$stateParams.productId, this.$scope.bdd.name)
-      .then((dumpIds) => dumpIds.map((id) => ({ id })))
-      .then((databaseDumps) => {
-        this.databaseDumps = databaseDumps;
-        return databaseDumps;
+    this.iceberg(
```
Comment (JacquesLarique) [inline packages/manager/apps/web/client/app/hosting/database/dump/hosting-database-dump.controller.js:55]: I think we should keep API calls inside services

**Review by JacquesLarique — APPROVED** [review]

**Review by qpavy — DISMISSED** [review]

**Review by JacquesLarique — APPROVED** [review]

**Pull request merged**